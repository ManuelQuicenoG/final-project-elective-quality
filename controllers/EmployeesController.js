import axios from 'axios';

export const employeeController = {
    register(employee) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: `employees`,
            data: employee,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'employees'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    },
    getemployee(name) {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    },
    updateemployee(name) {
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    }
}

