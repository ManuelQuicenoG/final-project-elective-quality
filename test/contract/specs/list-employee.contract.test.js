import { Matchers } from '@pact-foundation/pact';
import { employeeController } from '../../../controllers';
import { provider } from '../config/initPact';

describe('Employee Service', () => {
    beforeAll(async() => {
        await provider.setup();
    });
    describe('When a request to list all employees is made', () => {
        beforeAll(async () => {
            await provider.addInteraction({
                uponReceiving: 'a request to list all employees',
                state: "has employees",
                withRequest: {
                    method: 'GET',
                    path: '/employees'
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike(
                        {
                            name: Matchers.like("manuel nuevo"),
                            rolPosition: Matchers.like("AdministraciÃ³n"),
                            gender: Matchers.like("Male"),
                            hasCourses: Matchers.boolean(true)
                        }, {min: 1}
                    )
                }
            });
        });

        test('should return the correct data', async () => {
            const response = await employeeController.list();
            expect(response.data).toMatchSnapshot();
            
            await provider.verify()
        });

        afterAll(async() =>await provider.finalize());
    });
});