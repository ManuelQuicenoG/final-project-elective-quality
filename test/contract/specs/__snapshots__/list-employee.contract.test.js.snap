// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Service When a request to list all employees is made should return the correct data 1`] = `
Array [
  Object {
    "gender": "Male",
    "hasCourses": true,
    "name": "manuel nuevo",
    "rolPosition": "Administraci√≥n",
  },
]
`;
